packages/data-context/src/codegen/code-generator.ts

```diff
@@ -159,3 +162,22 @@ async function fileExists (absolute: string) {
     return false
   }
 }
+
+export async function hasNonExampleSpec (testTemplateDir: string, specs: string[]): Promise<boolean> {
```

Comment:
This code feels like it doesn't belong here and would make more sense in `ProjectActions`

Reply:
I think @warrensplayer had a reason for keeping it here & calling it from `ProjectActions`, maybe he can refresh my memory?

Reply:
@ZachJW34 @marktnoonan I felt a little weird about putting it there, but kept it in this file because I was using internal methods (`allFilesInDir` and `fileExists`) that I did not want to refactor at the time I created that method.

Reply:
I'm fine with where it is then, could do with a comment saying a refactor/move would be good.